tests/src/Kernel/AccessTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_contact_forms\Kernel;
+
+use Drupal\contact\Entity\ContactForm;
+use Drupal\Core\Access\AccessResult;
+
+/**
+ * Test the contact form access control.
+ */
+class AccessTest extends ContactFormTestBase {
+
+  /**
+   * Tests corporate contact form access.
+   */
+  public function testCorporateFormAccess(): void {
+    $contact_form = ContactForm::create(['id' => 'oe_contact_form']);
+    $contact_form->setThirdPartySetting('oe_contact_forms', 'is_corporate_form', TRUE);
+    $contact_form->save();
+
+    // Assert user with "access corporate contact form" permission has access
+    // to contact_form entity.
+    $account = $this->createUser(['name' => 'test1'], ['access corporate contact form']);
+    \Drupal::currentUser()->setAccount($account);
```

Comment:
Why are you setting the current user account. Just pass the created account to the access message.

Reply:
it complained about proxy, must have been something in dev stage, fixed